src/woo_search/search_index/tests/test_tasks.py

```diff
@@ -137,6 +140,324 @@ def test_index_document_roundtrip(self):
                 datetime(2030, 1, 5, 12, 0, 0, tzinfo=timezone.utc),
             )
 
+    @requests_mock.Mocker()
+    def test_full_text_updload(self, m):
+        with self.subTest("Happy flow"):
+            m.register_uri(
+                requests_mock.ANY, requests_mock.ANY, content=b"hello world."
+            )
+            document_uuid = "e90b8ea2-1ac2-4ef9-80ed-059d69eb3c54"
+
+            index_document(
+                uuid=document_uuid,
+                publicatie="d481bea6-335b-4d90-9b27-ac49f7196633",
+                informatie_categorieen=[
+                    {"uuid": "3c42a70a-d81d-4143-91d1-ebf62ac8b597", "naam": "WOO"}
+                ],
+                publisher={
+                    "uuid": "f8b2b355-1d6e-4c1a-ba18-565f422997da",
+                    "naam": "Utrecht",
+                },
+                identifier="https://www.example.com/1",
+                officiele_titel="A test document",
+                verkorte_titel="A document",
+                omschrijving="Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+                creatiedatum=date(2026, 1, 1),
+                registratiedatum=datetime(2026, 1, 5, 12, 0, 0, tzinfo=timezone.utc),
+                laatst_gewijzigd_datum=datetime(
+                    2026, 1, 5, 12, 0, 0, tzinfo=timezone.utc
+                ),
+                download_url="https://www.my_document.com/downloads/1",
+            )
+
+            # verify that it's indexed
+            with get_client() as client:
+                doc = Document.get(
+                    using=client,
+                    id=document_uuid,
+                )
+
+            # helps with type narrowing :)
+            assert isinstance(doc, Document), "Expected doc to be indexed"
+            self.assertEqual(doc.attachment.content, "hello world.")
+
+        with self.subTest(
+            "Download url with no content doesn't create attachment field in index."
+        ):
+            m.register_uri(requests_mock.ANY, requests_mock.ANY, status_code=204)
```

Comment:
I think you probably have to reset mocks in-between, I'm not sure what the behaviour is if there are multiple mocks matching.

Though I'd almost opt for a docker-compose service with a small flask app that performs real requests, and rely on VCR for all this stuff. There are examples of this in Open Forms.